### ISSUE TITLE
Failed to auto load specific file in phar

### ISSUE THREAD

**Issue by nsaumini** [composer/composer#6492]
My `composer.json`:

```json
"autoload": {
        ...
        "files" : [
        	"/../v1.0/Common.phar/index.php",
        ]
    }
```

Output of `composer diagnose`:

```
Checking composer.json: OK
Checking platform settings: OK
Checking git settings: OK
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com rate limit: OK
Checking disk free space: OK
Checking pubkeys:
Tags Public Key Fingerprint: 57815BA2 7E54DC31 7ECC7CC5 573090D0  87719BA6 8F3BB723 4E5D42D0 84A14642
Dev Public Key Fingerprint: 4AC45767 E5EC2265 2F0C1167 CBBB8A2B  0C708369 153E328C AD90147D AFE50952
OK
Checking composer version: OK
```

When I run this command:

```
composer update --no-dev
```

I get the following output:

```
Fatal error: composerRequire5270f4586e2a9545351a615f5264b486(): Failed opening required '/usr/lib/php5/voice/v1.12/vendor/composer/../../../../v1.0/Common.phar/index.php' (include_path='.:/usr/share/php:') in /usr/lib/php5/voice/vMultilang/vendor/composer/autoload_real.php on line 66
```

And I expected this to happen: No errors to come.

**Comment by nsaumini** [comment]
The issue is because inside `autoload_files.php` instead of `'2215619174705b9f4ed9b793bd371efe' => 'phar://' . $baseDir . '/../../v1.0/Common.phar/index.php'` it has following line `'2215619174705b9f4ed9b793bd371efe' => $baseDir . '/../../v1.0/Common.phar/index.php'`

**Comment by nsaumini** [comment]
Noticed following commit `https://github.com/composer/composer/commit/1f4882a3e6a229ceb878eba74e6a805cad086bf8`. Added a comment in there.

**Comment by alcohol** [comment]
@chuangbo @seldaek seems like we have a regression here. Maybe this particular bit of code needs some documentation/test so that we don't forget its purpose in the future?

**Comment by Seldaek** [comment]
Yeah that's why I was reluctant to change this as it was probably there for a reason.. The question is why this breaks, and can we maybe just add phar:// prefix but still use a relative path? I'll try something out quickly.

**Comment by Seldaek** [comment]
Soo what I don't understand is how this could have possibly worked before, as `preg_match('/\.phar$/', $path)` would not have matched this `.phar/index.php` path.

**Comment by Seldaek** [comment]
Merging the fix because it seems to work fine for me, but I am kinda confused how this was not a problem before and now it is..

**Comment by nsaumini** [comment]
@Seldaek I doesn't work for `.phar/index.php path.`, I come across this need only this week. I'm wondering if we can update the pattern to `'/\.phar.+$/'` so it will fix this issue. Then I noticed the `preg_match` line is removed in above mentioned commit.

At the end I just want to know how can I include a specific file in phar file under `files` option in `autoload` ?

**Comment by nsaumini** [comment]
@Seldaek Just only notice the commit. Thanks

**Comment by chuangbo** [comment]
@Seldaek @alcohol The path `.phar/index.php` doesn't work before nor after the fix #6378.

The new added `/\.phar.+$/` generates absolute path in `autoload_static.php` for phar files back again.

**Comment by alcohol** [comment]
A relative path should work too. The issue is that it simply cannot load the file inside the phar if the path is not prepended with `phar://` to indicate that it needs to use the phar wrapper.
